### PR TITLE
Initial checkin for a module for tolerant comparison of reals

### PR DESCRIPTION
**Pull request by arjenmarkus**
The module is based on the idea to allow for a small margin when comparing reals. Such a margin can solve the problem that the finite precision of floating-point calculations make the results inaccurate. Two essentially equal real numbers may therefore appear to be different. This is in response to issue #327

### CODE REVIEW

File: src/CMakeLists.bak [hunk 1]
```diff
@@ -0,0 +1,73 @@
+### Pre-process: .fpp -> .f90 via Fypp
```
Comment (awvwgk) [inline src/CMakeLists.bak:1]: `src/CMakeLists.bak` is not supposed to be committed here, is it?

**Comment by arjenmarkus** [comment]
Certainly not - github desktop is all too keen on submitting all files
residing in the checkout directory.

Op za 20 mrt. 2021 om 12:42 schreef Sebastian Ehlert <
***@***.***>:

**Comment by milancurcic** [comment]
Thanks @arjenmarkus, I think these will be useful. 

Is "tolerant" a commonly and uniquely used term? I wonder if most people would understand what it is just based on that word.

What do you think about `stdlib_comparisons`? But then it misses the point of the comparisons being tolerant. And `stdlib_tolerant_comparisons` is unwieldy.

I guess my main gripe with the word "tolerant" is that it's an adjective and not a noun, i.e. it doesn't say what is in the module, but what something is like in the module.

**Comment by arjenmarkus** [comment]
The original source code mentions "fuzzy Fortran operators", that may be
more apt, but then I fear people would expect fuzzy sets and the like. I am
not too happy with the term tolerant either, so I am open to suggestions.

Op ma 22 mrt. 2021 om 20:46 schreef Milan Curcic ***@***.***>:

**Comment by milancurcic** [comment]
Currently `stdlib_comparison` or `stdlib_comparisons` is my favorite. We can request feedback on the call tomorrow.

**Comment by arjenmarkus** [comment]
The plural has my preference. As I will not be able to join today, I will
simply await the results.

Op di 23 mrt. 2021 om 23:35 schreef Milan Curcic ***@***.***>:

**Review by jvdp1 — COMMENTED** [review]
Thank ypou @arjenmarkus for this PR. Here are some comments.
I agree with @milancurcic: `stdlib_comparison` or `stdlib_comparisons` would be better thatn `stdlib_tolerant`. However, they don't cover `tfloor`,,... What do you think about `stdlib_operators`?

General comments:
* Some hyperlinks mst still be added in the code, but this could be done at the end of the PR
* `fypp` could be used to generated the code for the different kinds of `real`.

**Comment by awvwgk** [comment]
With https://github.com/fortran-lang/stdlib/pull/488 this functionality was implemented, therefore I'm going to close this PR.

Feel free to reopen this PR if you think #488 doesn't address all changes included in this patch.

**Pull request closed without merging**